std/typecons.d

```diff
@@ -3987,10 +4027,21 @@ if (Targets.length >= 1 && allSatisfy!(isMutable, Targets))
             this(       inout Source s)        inout @safe pure nothrow { _wrap_source = s; }
             this(shared inout Source s) shared inout @safe pure nothrow { _wrap_source = s; }
 
-            // BUG: making private should work with NVI.
-            protected final inout(Object) _wrap_getSource() inout @trusted
+            static if(is(Source == class) || is(Source == interface))
             {
-                return dynamicCast!(inout Object)(_wrap_source);
+                // BUG: making private should work with NVI.
+                protected final inout(Object) _wrap_getSource() inout @safe
+                {
+                    return dynamicCast!(inout Object)(_wrap_source);
+                }
+            }
+            else
+            {
+                // BUG: making private should work with NVI.
+                protected final inout(Source) _wrap_getSource() inout @safe
```

Comment:
@andralex, This particular piece of code is from the original implementation. When utilizing a struct copies are made, I don't see a way to store it as a reference or unwrap as a reference. I'll add this to the documentation.